conan/cli/commands/workspace.py

```diff
@@ -237,6 +230,91 @@ def workspace_init(conan_api: ConanAPI, parser, subparser, *args):
+    exported_refs = conan_api.workspace.export()
+
+    build_mode = args.build if args.build else []
+    build_mode.extend(f"missing:{r}" for r in exported_refs)
+
+    all_packages = conan_api.workspace.editable_packages()
+    packages = conan_api.workspace.select_packages(args.pkg)
+
+    # If we don't disable the workspace, then, the packages are not created in the Conan cache,
+    # but locally in the user folders, are they are intercepted as editables
+    conan_api.workspace.enable(False)
+
+    install_order = conan_api.workspace.build_order(packages, profile_host, profile_build, build_mode,
+                                                    lockfile, remotes, args, update=args.update)
+
+    ConanOutput().box("Workspace creating each package")
+    order = install_order.install_build_order()
+
+    profile_args = ProfileArgs.from_args(args)
+    for level in order["order"]:
+        for elem in level:
+            ref = RecipeReference.loads(elem["ref"])
+            for package_level in elem["packages"]:
+                for package in package_level:
+                    if package["binary"] != "Build":
+                        continue
+
+                    if ref not in all_packages:
+                        # Build external to Workspace
+                        cmd = f'install {package["build_args"]} {profile_args}'
+                        ConanOutput().box(f"Workspace building external {ref}")
+                        ConanOutput().info(f"Build command: {cmd}\n")
+                        conan_api.command.run(cmd)
+                    else:  # Package in workspace
+                        path = packages[ref]["path"]
+                        # TODO: Missing --lockfile-overrides arg here
+                        build = "--build-require" if package["context"] == "build" else ""
+                        ref_args = " ".join(f"--{k}={getattr(ref, k)}"
+                                            for k in ("name", "version", "user", "channel")
+                                            if getattr(ref, k, None))
+                        cmd = f'create "{path}" {profile_args} {build} {ref_args}'
+                        ConanOutput().box(f"Workspace create {ref}")
+                        ConanOutput().info(f"Conan create command: {cmd}\n")
+                        conan_api.command.run(cmd)
+
+
+@conan_subcommand()
+def workspace_source(conan_api: ConanAPI, parser, subparser, *args):
+    """
+    Call the source() method of packages in the workspace
+    """
+    subparser.add_argument("--pkg", action="append", help='Define specific packages')
+    args = parser.parse_args(*args)
+
+    remotes = conan_api.remotes.list()  # In case "python_requires" are needed
+    packages = conan_api.workspace.select_packages(args.pkg)
+
+    ConanOutput().box("Workspace getting sources")
+    for pkg, info in packages.items():
+        conan_api.local.source(info["path"], name=pkg.name, version=pkg.version,
```

Comment:
```suggestion
        conan_api.local.source(info["path"], name=pkg.name, version=str(pkg.version),
```
confirmed that this would cause issues, as version is a `Version` which would end up as the `conanfile.version` in `load_named`, thus making the `conanfile.source()` method call fail if iot uses the `self.version` expecting it to be a str as elsewhere